app/services/calculators/diapers_service.rb

```diff
@@ -0,0 +1,56 @@
+class Calculators::DiapersService
+  @@diapers_per_month = {
+    (1..3) =>
+      { price: 1220,
+        amount: 305 },
+    (4..6) =>
+      { price: 1098,
+        amount: 244 },
+    (7..9) =>
+      { price: 915,
+        amount: 183 },
+    (10..12) =>
+      { price: 1006.5,
+        amount: 183 },
+    (13..18) =>
+      { price: 671,
+        amount: 122 },
+    (19..24) =>
+      { price: 732,
+        amount: 122 },
+    (25..30) =>
+      { price: 366,
+        amount: 61 }
+  }.freeze
+
+  attr_reader :age, :used_diapers_amount, :to_be_used_diapers_amount,
+              :used_diapers_price, :to_be_used_diapers_price
+
+  def initialize(age)
+    @age = age
+    @used_diapers_amount = 0
+    @to_be_used_diapers_amount = 4575
+    @used_diapers_price = 0
+    @to_be_used_diapers_price = 23_332.5
+  end
+
+  def calculate!
+    @@diapers_per_month.each_key do |key|
+      if key.last <= @age
+        change(key, key.size)
+      elsif key.include? @age
+        change(key, @age - key.first + 1)
+      end
+    end
+    @to_be_used_diapers_amount -= @used_diapers_amount
```

Comment:
No need to call @.. vars if you add it to `attr_reader `
```suggestion
    @to_be_used_diapers_amount -= used_diapers_amount
```